wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt

```diff
@@ -654,7 +654,15 @@ class KotlinGenerator private constructor(
     return when {
       isScalar -> CodeBlock.of("%T.%L", ProtoAdapter::class, simpleName().toUpperCase(Locale.US))
       isMap -> throw IllegalArgumentException("Can't create single adapter for map type $this")
-      else -> CodeBlock.of("%L.ADAPTER", typeName.simpleName)
+      else -> CodeBlock.of("%L.ADAPTER", typeName.getQualifiedName())
```

Comment:
I think using `%T` here instead of `%L` is the right thing to do. This will automatically figure out the shortest unambiguous name to qualify the type.